RPC.en.rst

```diff
@@ -3,59 +3,257 @@
-      root of the assymmetry. See :ts:git:`mgmt/api/NetworkMessage.cc`.
+.. class:: LocalManager : BaseManager
 
-   #. If this is made in to a library, it will be necessary to make the registration of message and
-      type handling a push mechanism rather than a push. That is, currently this information is
-      directly embedded in the RPC code, which won't be possible in a library.
+   This class is used by ``traffic_manager`` to communicate with ``traffic_server``
 
-   #. Make it possible for the core to pass messages back from ``traffic_server`` to command line tools.
+.. function:: void LocalManager::pollMgmtProcessServer()
 
-   #. Make it possible for plugins to pass messages back to command line tools.
+   This function watches the socket and handles incoming messages from processes. Used in the main event loop of ``traffic_manager``.
 
-#. Create ``libatsprc``.
+.. function:: void LocalManager::sendMgmtMsgToProcesses( ... )
 
-   a. Create the library.
+   This function is used by ``traffic_manager`` to process the messages based on msg_id. It then sends the message to ``traffic_server`` over sockets.
 
-   #. Change the RPC using code to use the library.
+.. class:: ProcessManager : BaseManager
+
+   This class is used by ``traffic_server`` to communicate with ``traffic_manager``
+
+.. function:: int ProcessManager::pollLMConnection()
+
+    This function periodically polls the socket to see if there are any messages from the ``LocalManager``. It can accept up to ``MAX_MSGS_IN_A_ROW``, defaulted at 10000. 
+
+.. function:: void ProcessManager::signalManager ( ... )
+
+   This function sends messages to the LocalManager using sockets. Details on how the write is preformed is in the section "Messages over sockets between traffic_manager and traffic_server".
 
 .. rubric:: Notes
 
-Variadic Template Support
-   I considered this for serialization but the problem is deserializing. Given a message type, how
-   is a call using variadics to be constructed? You would still need some sort of registration for a
-   functor to invoke, at which point it's not really winning. There would also be the problem of
-   matching serialized type data with C++ types, again obviating the advantages.
+1. In the context of the ``ProcessManager``, the ``mgmt_signal_queue`` are signals to be sent from ``traffic_server`` to ``traffic_manager``. The ``mgmt_event_queue`` are events and signals recieved from ``traffic_manager`` to ``traffic_server``. This distinction is important as ``processSignalQueue( ... )`` and ``processEventQueue( ... )`` are preforming different tasks.
+
+#. Both ``pollMgmtProcessServer()`` and ``pollLMConnection()`` actually use ``select()``, not ``poll()`` or ``epoll()``, underneath.
+
+
+Messages over sockets between ``traffic_manager`` and ``traffic_server``
+========================================================================
+
+Read:
+
+1. Read the MgmtMessageHdr:
```

Comment:
We need to have you play with plantUML for this. I'll see if I can do an example.

Reply:
added ====== to this